framework/src/modules/chain/bft/bft.js

```diff
@@ -38,8 +40,9 @@ const validateBlockHeader = blockHeader =>
  * @property {string} delegatePublicKey
  */
 
-class BFT {
+class BFT extends EventEmitter {
```

Comment:
This is the general pattern we follow in our application, I am merely building on top of the existing event declaration/handling style.

Reply:
Secondly, the inheritance of BFT class and Delegate class from EventEmitter in this PR is merely to declare (without runtime errors) the integration logic in the chain module for event handlers in dpos. The proper system and structure for event emitters can be discussed in an STM. Would you agree?

Reply:
We can have an STM but as i said earlier, I have a strong opinion on this. Other devs may repeat the same pattern and it's just more work for future. However, i don't want to prevent the PR to be merged. This can be easily refactored. If we merge this PR like this, would be great if we could have an issue to follow up on this.

Reply:
I created issue #3904 to further discuss this point.

Reply:
@SargeKhan could you please the issue and reference it to this discussion?